helix-view/src/document.rs

```diff
@@ -236,26 +242,107 @@ impl Document {
     }
 
     // TODO: async fn?
-    pub fn load(
+    fn open_impl(
+        path: PathBuf,
+        encoding: Option<&'static encoding_rs::Encoding>,
+    ) -> Result<(Rope, &'static encoding_rs::Encoding), Error> {
+        use std::{fs::File, io::Read};
+
+        if !path.exists() {
+            return Err(anyhow::anyhow!("path doesn't exist"));
+        }
+
+        // Create two 8kB buffers to write the file into and the decoded contents out to.
+        let mut buf = [0u8; BUF_SIZE];
+        let mut buf_out = [0u8; BUF_SIZE];
```

Comment:
Do we want to use `MaybeUninit` here?

Reply:
I don't know any `unsafe`, so I think someone else would have to evaluate that I think.

Reply:
I think the code as-is is simpler.  Technically we're taking a small performance hit for zeroing out the memory, but it's only done once at the beginning.  And given that this function is making system calls to read from the file system (which is orders of magnitude slower), I don't think it's something we need to worry about here.